src/cookbook/ScottPlot.Cookbook/Locate.cs

```diff
@@ -10,66 +10,96 @@ namespace ScottPlot.Cookbook
 {
     public static class Locate
     {
-        public static IRecipe[] GetRecipes() =>
-            AppDomain
-            .CurrentDomain
-            .GetAssemblies()
+        private static readonly IRecipe[] _recipes = AppDomain.CurrentDomain.GetAssemblies()
             .SelectMany(s => s.GetTypes())
             .Where(x => x.IsAbstract == false)
             .Where(x => x.IsInterface == false)
             .Where(p => typeof(IRecipe).IsAssignableFrom(p))
             .Select(x => (IRecipe)Activator.CreateInstance(x))
             .ToArray();
 
-        public static IRecipe GetRecipe(string id) => GetRecipes().Where(x => x.ID == id).First();
+        private static readonly string[] _categories = _recipes.Select(r => r.Category).Distinct().ToArray();
+        private static Dictionary<string, IRecipe[]> _recipesByCategory = new Dictionary<string, IRecipe[]>();
+        private static Dictionary<string, IRecipe> _recipesById = new Dictionary<string, IRecipe>();
+        private static List<KeyValuePair<string, IRecipe[]>> _categorizedRecipeList = new List<KeyValuePair<string, IRecipe[]>>();
 
-        public static string[] GetCategories() => GetRecipes().Select(x => x.Category).Distinct().ToArray();
-
-        public static string[] GetCategoriesInDisplayOrder() => GetCategorizedRecipes().Select(x => x.Key).ToArray();
+        private static readonly string[] topCategories =
+        {
+            "Quickstart",
+            "Axis and Ticks",
+            "Advanced Axis Features",
+            "Multi-Axis",
+        };
 
-        public static IRecipe[] GetRecipes(string category) => GetRecipes().Where(x => x.Category == category).ToArray();
+        private static readonly string[] bottomCategories =
+        {
+            "Style",
+            "Misc"
+        };
 
-        public static List<KeyValuePair<string, IRecipe[]>> GetCategorizedRecipes()
+        static Locate() // A static constructor runs exactly once and before the class or an instance of it is needed
         {
-            var categorizedRecipeList = new List<KeyValuePair<string, IRecipe[]>>();
+            InitializeDictionaries();
+            InitializeCategoryList();
+        }
 
-            foreach (string category in GetCategories())
+        private static void InitializeDictionaries()
+        {
+            Dictionary<string, List<IRecipe>> byCategory = new Dictionary<string, List<IRecipe>>();
```

Comment:
This dictionary can be initialized easier using LINQ:
```c#
Dictionary<string, List<IRecipe>> byCategory = GetRecipes.GroupBy( x => x.Category).ToDictionary(  gk => gk.Key, gk => gk.ToList());
```

Reply:
Cool, I hadn't known about this. Reminds me of dictionary comprehension in Python.